contracts/token/erc20/preset/ImmutableERC20MinterBurnerPermit.sol

```diff
@@ -0,0 +1,70 @@
+// Copyright (c) Immutable Pty Ltd 2018 - 2024
+// SPDX-License-Identifier: MIT
+pragma solidity 0.8.19;
+
+import {ERC20Permit, ERC20} from "@openzeppelin/contracts/token/ERC20/extensions/ERC20Permit.sol";
+import {ERC20Burnable} from "@openzeppelin/contracts/token/ERC20/extensions/ERC20Burnable.sol";
+import {ERC20Capped} from "@openzeppelin/contracts/token/ERC20/extensions/ERC20Capped.sol";
+import {MintingAccessControl, AccessControl, IAccessControl} from "../../../access/MintingAccessControl.sol";
+import {IImmutableERC20Errors} from "./Errors.sol";
+
+/**
+ * @notice ERC 20 contract that wraps Open Zeppelin's ERC 20 contract.
+ * This contract has the concept of an owner, called _owner in the constructor.
+ * This account has no rights to execute any administrative actions within the contract,
+ *  with the exception of transferOwnership and role grants/revokes. This account is accessed via the owner() 
+ *  function. The Immutable Hub uses this function to help associate the ERC 20 contract 
+ *  with a specific Immutable Hub account.
+ */
+contract ImmutableERC20MinterBurnerPermit is ERC20Capped, ERC20Burnable, ERC20Permit, MintingAccessControl {
+    /// @notice Role to mint tokens
+    bytes32 public constant HUB_OWNER_ROLE = bytes32("HUB_OWNER_ROLE");
+
+    /**
+     * @dev Delegate to Open Zeppelin's contract.
+     * @param _name  Name of the token.
+     * @param _symbol Token symbol.
+     * @param _hubOwner The account that owns the contract and is associated with Immutable Hub. 
+     * @param minterRole The account that has the MINTER_ROLE.
+     * @param _maxTokenSupply The maximum supply of the token.
+     * @param _admin The account that has the DEFAULT_ADMIN_ROLE.
+     */
+    constructor(string memory _name, string memory _symbol, address _hubOwner, address minterRole, uint256 _maxTokenSupply, address _admin) ERC20(_name, _symbol) ERC20Permit(_name) ERC20Capped(_maxTokenSupply) {
```

Comment:
I would order the parameters:
_roleAdmin
_minterAdmin
_hubOwner
_name
_symbol
_maxTokenSupply

That is, order all of the admin roles together and all of the token stuff together.